### PR TITLE
Storage pool absent

### LINKED ISSUE #5775: KVM: probable problem with storage stats when storage is put on maintenance
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Primary storage - excluding Ceph/non managed NFS
~~~

##### CLOUDSTACK VERSION

~~~
main and older versions
~~~

##### OS / ENVIRONMENT

OS - CentOS7
hypervisor - KVM
mgmt/agent version - 4.16.0.0

##### SUMMARY

When you put the primary storage on Maintenance mode and restart the agent, the storage will be removed from the storage adaptor's map. This causes NullPointerException when collecting the storage statistics. The problem is probably for all storage pools which don't use `LibvirtStorageAdaptor` - e.g. Linstor, StorPool maybe, and SolidFire.
This does not affect Ceph and non managed NFS, also and PowerFlex (PowerFlex does not call the agents to get the storage statistics)

##### STEPS TO REPRODUCE

~~~
set storage.stats.interval on short interval
enable storage maintenance mode
restart the agents
~~~

##### EXPECTED RESULTS

~~~
collect or don't the storage stats on pools on Maintenance mode
~~~

##### ACTUAL RESULTS

~~~
2021-12-15 08:31:03,586 DEBUG [cloud.agent.Agent] (agentRequest-Handler-3:null) (logid:552801e3) Processing command: com.cloud.agent.api.GetStorageStatsCommand
2021-12-15 08:31:03,586 WARN  [cloud.agent.Agent] (agentRequest-Handler-3:null) (logid:552801e3) Caught: 
java.lang.NullPointerException
        at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetStorageStatsCommandWrapper.execute(LibvirtGetStorageStatsCommandWrapper.java:40)
        at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetStorageStatsCommandWrapper.execute(LibvirtGetStorageStatsCommandWrapper.java:32)
        at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtRequestWrapper.execute(LibvirtRequestWrapper.java:78)
        at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.executeRequest(LibvirtComputingResource.java:1768)
        at com.cloud.agent.Agent.processRequest(Agent.java:661)
        at com.cloud.agent.Agent$AgentRequestHandler.doTask(Agent.java:1079)
        at com.cloud.utils.nio.Task.call(Task.java:83)
        at com.cloud.utils.nio.Task.call(Task.java:29)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
2021-12-15 08:31:03,588 DEBUG [cloud.agent.Agent] (agentRequest-Handler-3:null) (logid:552801e3) Seq 60-4602115869219225621:  { Ans: , MgmtId: 2199191159196, via: 60, Ver: v1, Flags: 10, [{"com.cloud.agent.api.Answer":{"result":"false","details":"java.lang.NullPointerException

~~~

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR Fixes #5775

Fixes: #5775

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@DaanHoogland can you rebase with 4.16 (and change the target to 4.16).

**Review by GabrielBrascher — APPROVED** [review]
Code LGTM

**Review by slavkap — APPROVED** [review]
code LGTM, also tested

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetStorageStatsCommandWrapper.java [hunk 1]
```diff
@@ -37,6 +37,9 @@ public Answer execute(final GetStorageStatsCommand command, final LibvirtComputi
         try {
             final KVMStoragePoolManager storagePoolMgr = libvirtComputingResource.getStoragePoolMgr();
             final KVMStoragePool sp = storagePoolMgr.getStoragePool(command.getPooltype(), command.getStorageId(), true);
+            if(null == sp) {
```
Comment (GutoVeronezi) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetStorageStatsCommandWrapper.java:40]:
```suggestion
            if (null == sp) {
```

**Review by GutoVeronezi — APPROVED** [review]
CLGTM

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetStorageStatsCommandWrapper.java [hunk 2]
```diff
@@ -37,6 +37,9 @@ public Answer execute(final GetStorageStatsCommand command, final LibvirtComputi
         try {
             final KVMStoragePoolManager storagePoolMgr = libvirtComputingResource.getStoragePoolMgr();
             final KVMStoragePool sp = storagePoolMgr.getStoragePool(command.getPooltype(), command.getStorageId(), true);
+            if (null == sp) {
```
Comment (rohityadavcloud) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetStorageStatsCommandWrapper.java:40]:
```suggestion
            if (sp == null) {
```

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**